Advertise MAM in disco#info

jzskca opened the pull request:
This PR addresses #2269.

Proposed changes include:
 * Advertise MAM for user JIDs.
 * Advertise MAM for MUC Light rooms.

As currently written, this unconditionally advertises MAM for MUC Light rooms, since MAM is a requirement for MUC Light.

fenek commented on src/muc_light/mod_muc_light_codec_modern.erl:
```diff
@@ -268,7 +268,12 @@ encode_iq({get, #disco_info{ id = ID }}, _RoomJID, _RoomBin, _HandleFun) ->
-                #xmlel{name = <<"feature">>, attrs = [{<<"var">>, ?NS_MUC_LIGHT}]}],
+                #xmlel{name = <<"feature">>, attrs = [{<<"var">>, ?NS_MUC_LIGHT}]},
+                % MUC Light requires MAM
+                #xmlel{name = <<"feature">>, attrs = [{<<"var">>, ?NS_MAM_03}]},
```
These shouldn't be advertised if MAM is not enabled.

jzskca commented:
Updated.

I removed the changes to `mod_mam` and just addressed advertising for MUC, since the source of that problem seems clear (`disco#info` is handled by the MUC modules directly instead of by `mod_disco`). For `disco#info` queries to users, although MAM is registered with `mod_disco`, `mod_disco:get_sm_features` does not return registered features, unlike `mod_disco:get_local_features`. I don't understand the distinction between the two well enough to offer a solution.

I haven't figured out yet why test steps 3, 7, and 8 are failing the `disco_features` tests. Any advice would be appreciated.

aaabbbccc-123 commented:
just FYI, src/mod_muc_room.erl need to be update with similar change to make converse.js show MAM support for the room. Reading from the specs, that should be the correct interpretation i think.
